_rules/block-collapsible-cefbef.md

```diff
@@ -0,0 +1,47 @@
+---
+id: cefbef
+name: Block of content is expandable and collapsible
+rule_type: atomic
+description: |
+  This rule checks that repeated blocks of content are expandable and collapsible
+accessibility_requirements:
+  wcag-technique:SCR28: # Using an expandable and collapsible menu to bypass block of content
+		forConformance: false
+		failed: not satisfied
+		passed: satisfied
+		inapplicable: further testing needed
+input_aspects:
+	- DOM Tree
+	- CSS styling
+authors:
+	- Jean-Yves Moyen
+	- Anne Thyme Nørregard
+---
+
+## Applicability
+
+This rule applies to any [document](#https://www.w3.org/TR/dom/#concept-document) where the [document element](#https://www.w3.org/TR/dom/#document-element) is an HTML `html` element.
+
+## Expectations
+
+For each [section of repeated content](#repeated-content) in the test target, there exists some [user interface component](https://www.w3.org/TR/WCAG21/#dfn-user-interface-components) that allows to toggle both [visibility](#visible) and [inclusion in the accessibility tree](#included-in-the-accessibility-tree) of this [section of repeated content](#repeated-content).

```

Comment:
The applicability has moved back and forth…

Ultimately, the problem with moving "section of repeated content" to the applicability is that the applicability of the composite rule has to be the union of the accessibilities of the atomic rules (per ACT-Rule format). Having the accessibility of the composite rule being "the document, and the sections of repeated content, and the first focusable element" makes it super messy. Especially, we can't anymore use the "for each target, one atomic rule must pass" composition since a page could have its first focusable failing/inapplicable for all atomic rules, but still be fine because it's actually using headings.

Reply:
But "the document" is already the union of the applicability of the atomic rules (i.e., all sections of content and focusable elements are in the document).

Reply:
I think the union has to be taken on the set of test targets. So the test targets of the composite rules are all the test targets for each atomic rule. Otherwise, we can't write the expectation of the composite rule as simple logic between the outcomes of the atomic rules.

That is, if one of the atomic rule has focusable elements as test target and the composite rule only has the full document, then any test target for the composite rule will be inapplicable for the atomic rule. So in order to use that atomic rule inside the composite rule, the composite rule must also accept focusable elements as test targets.

Pushing the example further, if atomic rule A1 targets focusable elements, and atomic rule A2 targets section of content, and composite rule C using only these two rules targets documents, then any test target for C will be inapplicable for both A1 and A2 and there is no way to combine the outcomes in any meaningful way (since the outcomes are not meaningful).

So, I understand when you say that a document is a union of its sections of content, but I don't think it is at that level that the union is taken. To put it in more formal term, a test target is a set (*e.g.* a subtree of the DOM seen as the set of nodes) but the applicability for composite rule makes the union of the sets of test targets, not the set of unions of test targets.

That is, if rule A1 has test targets `{1, 2}` and `{2, 3}`, then its applicability is `{ {1, 2} , {2, 3} }`; and if rule A2 has applicability `{ {2, 4} , {3, 4} }`; then the applicability of a composite rule should be `{ {1, 2} , {2, 3} , {2, 4} , {3, 4} }` and not, say `{ {1, 2, 4} , {2, 3, 4} }` or `{ {1, 2, 4} , {1, 3, 4} , {2, 3, 4} }`, or …

Reply:
I'm not sure what is expressed in https://www.w3.org/TR/act-rules-format/#applicability-composite is to be taken so literally. @WilcoFiers could you help with the interpretation of the ACT Rules spec?
The ACT rules format allows authors to omit the applicability of composite rules. We could use that here (just because I believe having a more precise applicability in the atomic rules would be helpful).

Reply:
I agree that more precise applicability in the atomic rules would be very helpful. This is what I did in commit 72ccf3c, you can for example see:
* [first focusable](https://github.com/act-rules/act-rules.github.io/blob/68424039f7ca22cb34b9b85d5f02797c18da0879/_rules/first-focusable-elements-are-links-to-content-e53727.md)
* [block collapsible](https://github.com/act-rules/act-rules.github.io/blob/68424039f7ca22cb34b9b85d5f02797c18da0879/_rules/block-collapsible-cefbef.md)
* [composite rule](https://github.com/act-rules/act-rules.github.io/blob/68424039f7ca22cb34b9b85d5f02797c18da0879/_rules/bypass-blocks-cf77f2.md)

But after talking with Anne, we agreed that this was not allowed by the format :-(